### PR TITLE
replace macros with functions

### PR DESCRIPTION
**Pull request by lucalewin**
pr for issue #128 
maybe take a look at ˋopen_elements_getˋ in ˋparser.rsˋ. Before this function the macro used ˋusizeˋ and ˋNodeIdˋ but now the function only take a ˋusizeˋ. I don't know if that really changes anything. I couldn't use just ˋNodeIdˋ because there is one place where we don't have a NodeId.

### CODE REVIEW

**Comment by jaytaph** [comment]
Nice PR! Indeed we don't need those macro's at this level, but this is mostly due to my inexperience with rust itself. 

When @emwalker @Kiyoshika approves, we can merge this

**Review by jaytaph — APPROVED** [review]

**Review by emwalker — APPROVED** [review]

File: src/html5_parser/parser.rs [hunk 1]
```diff
@@ -1687,6 +1532,128 @@ impl<'a> Html5Parser<'a> {
         ))
     }
 
+    fn acknowledge_closing_tag(&mut self, is_self_closing: bool) {
+        if is_self_closing {
+            self.ack_self_closing = true;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jaytaph) [reply src/html5_parser/parser.rs:1539]: ack_self_closing might altready be true, but the is_self_closing could be false. In that case, ack-self_closing must stay on true

[the remark this answers is not in the document]
Comment (Kiyoshika) [reply src/html5_parser/parser.rs:1539]: Oh gotcha, that makes sense

**Comment by jaytaph** [comment]
@lucalewin thanks for the pr!

**Comment by lucalewin** [comment]
I didn't check the benchmarks for these changes. It could maybe be slower now because the macro basically inserted the code directly but now we have a function call which can be slower. If that is the case we can add `#[inline(always)]` to the functions. Then the compiler is forced to inline the function even in dev mode. In release mode this should automatically be inlined as an optimization by the compiler.

**Comment by emwalker** [comment]
I ran benchmarks before and after your PR.  I'd say the difference is close to being hard to measure, given the variance I've seen in my own benchmark runs over the last two weeks.  These are the third run of three runs for each case, so six runs altogether, four for warming up.





A week or so ago I was regularly seeing greater than 6ms for the run.  So I would not worry about this apparent difference.  I suspect there are OOM improvements that we could tackle first before looking at this level of change.

**Pull request merged**